web/app/datasets/test_handle_signals.py

```diff
@@ -0,0 +1,48 @@
+import json
+import datetime
+
+from django.test import TestCase
+from django.utils import timezone
+from django.test.utils import override_settings
+import pytz
+
+from datasets import handle_signals
+from datasets.models import MessageLog
+from datasets.external.base import reset_handlers
+
+SIGNAL_PLACEHOLDERS = json.loads("""
+[
+    {
+        "signal_id": "1"
+    },
+    {
+        "signal_id": "2"
+    },
+    {
+        "signal_id": "4"
+    }
+]
+""")
+
+
+class TestSignalHandling(TestCase):
+    def setUp(self):
+        reset_handlers()
+        MessageLog.objects.create(
+            signal_id='1',
+            t_entered=timezone.now(),
+            t_sent=timezone.now(),
+            handler_name='some-handler',
+            status='200',
+            is_sent=True
+        )
+
+    def test_call_external_apis(self):
+        signals = SIGNAL_PLACEHOLDERS
```

Comment:
This is a normal flow test. The test should actually test the borders, empty list, list with one element, list with more elements (as you have), trying to write the same list twice, trying to write a new list with one already written signal, etcetera. Actually, almost only things that in practice are not to be expected to happen.

Reply:
Will take this into account for the next iteration of tests. (Want to really test this service well.)

Reply:
OK, point taken